validator/src/journal/block_manager.rs

```diff
@@ -557,20 +557,59 @@ impl Iterator for GetBlockIterator {
 
 pub struct BranchIterator {
     state: Arc<RwLock<BlockManagerState>>,
+    initial_block_id: String,
     next_block_id: String,
     blockstore: Option<String>,
 }
 
 impl BranchIterator {
     fn new(state: Arc<RwLock<BlockManagerState>>, first_block_id: String) -> Self {
+        let next_block_id = {
+            let mut block_manager = state
+                .write()
+                .expect("No log holder should have poisoned the lock");
+            match block_manager.ref_block(&first_block_id) {
+                Ok(_) => first_block_id.clone(),
+                Err(BlockManagerError::UnknownBlock) => NULL_BLOCK_IDENTIFIER.to_string(),
+
+                Err(err) => {
+                    error!(
+                        "Unable to ref block at {}: {:?}; ignoring",
+                        &first_block_id, err
+                    );
+                    NULL_BLOCK_IDENTIFIER.to_string()
+                }
+            }
+        };
         BranchIterator {
             state,
-            next_block_id: first_block_id,
+            initial_block_id: first_block_id.clone(),
+            next_block_id,
             blockstore: None,
         }
     }
 }
 
+impl Drop for BranchIterator {
+    fn drop(&mut self) {
+        let mut block_manager = self.state
+            .write()
+            .expect("No log holder should have poisoned the lock");
```

Comment:
The argument should state the error that occurred, not the expected behavior. Also there is a typo.